### ISSUE TITLE
placeholder setting an array with a sequence

### ISSUE THREAD

**Issue by inksci** [tensorflow/tensorflow#41655]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1.  It must be a bug, a feature request, or a significant problem with the
    documentation (for small docs fixes please send a PR instead).
2.  The form below must be filled out.
3.  It shouldn't be a TensorBoard issue. Those go
    [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**:
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**:
-   **TensorFlow installed from (source or binary)**:
-   **TensorFlow version (use command below)**:
-   **Python version**:
-   **Bazel version (if compiling from source)**:
-   **GCC/Compiler version (if compiling from source)**:
-   **CUDA/cuDNN version**:
-   **GPU model and memory**:
-   **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by amahendrakar** [comment]
@inksci,
Could you please elaborate your concern and describe the issue you are facing?

Also in order to expedite the trouble-shooting process, please fill in the issue template. Thanks!

**Comment by inksci** [comment]
there is a placeholder: shape=(None,None,3)
for the input: both x1 with shape (1,5,3) and x2 with shape (1,4,3) can feed into the network and get the output.
In fact, conv1d and global max_pooling is used in my networks to deal with these inputs with changed shape (at axis=1).
However, when training the network, for example the batch_size is 2 and the input data 
[x1, x2 ]
which feeded into the placeholder with shape of (None, None, 3)
You know, it will make a error: setting an array with a sequence.
The problem is that x1 and x2 have different domension. I know the problem but how can I train the network with datasets which contains data have different dimensions.
You know, if I feed these data one by one into networks, that is ok.
But for a batch training, these data with different domension can not accepted by placeholder except the batch_size is 1.
how can I training the networks with these irregular input data? will the tf.keras or tf2 can offer a solution?
Thanks a lot, and hope that I express my meaning clearly.

**Comment by amahendrakar** [comment]
> However, when training the network, for example the batch_size is 2 and the input data
> [x1, x2 ]
> which feeded into the placeholder with shape of (None, None, 3)
> You know, it will make a error: setting an array with a sequence.

@inksci,
Could you please provide the complete code and the dataset you are using, to reproduce the issue reported here. Thanks!
